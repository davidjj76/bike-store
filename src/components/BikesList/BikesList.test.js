import React from 'react';
import { shallow } from 'enzyme';
import BikesList from './BikesList';

describe('BikeList', () => {
  const props = {
    className: 'test',
    bikes: [{ id: '1' }, { id: '2' }],
    addToCart: jest.fn(),
  };
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<BikesList {...props} />);
  });

  test('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should have a "test" class', () => {
    expect(wrapper.hasClass(props.className)).toBe(true);
  });

  test('should render a List of bikes', () => {
    expect(wrapper.find('List').props().items).toEqual(props.bikes);
  });

  test('should add a bike to cart', () => {
    const bike = {
      id: '1',
      price: 895,
      name: 'name',
      image: 'image',
      hasStock: true,
    };
    const bikeItem = shallow(wrapper.find('List').props().renderItem(bike));
    bikeItem.find('button').simulate('click');
    expect(props.addToCart).toHaveBeenCalledWith(bike.id);
    // const bikeItem = wrapper.find('List').props().renderItem(bike);
    // bikeItem.props.onAddToCartClick();
    // expect(props.addToCart).toHaveBeenCalledWith(bike.id);
  });
});
