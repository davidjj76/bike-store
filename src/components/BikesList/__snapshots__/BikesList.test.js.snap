// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeList snapshot testing 1`] = `
<div
  className="bikes-list test"
>
  Bikes List
  <List
    className="list"
    items={
      Array [
        Object {
          "id": "1",
        },
        Object {
          "id": "3",
        },
      ]
    }
    renderItem={[Function]}
  />
</div>
`;
